plugins/org.komodo.core/src/org/komodo/repository/ObjectImpl.java

```diff
@@ -583,6 +583,16 @@ public KomodoObject getChild( final UnitOfWork transaction,
     /**
      * {@inheritDoc}
      *
+     * @see org.komodo.spi.repository.KomodoObject#getChildTypes()
+     */
+    @Override
+    public KomodoType[] getChildTypes() {
+        return KomodoType.NO_TYPES;
+    }
```

Comment:
The javadocs for that method state that if an empty array is returned a child of any type can be created. The intent is this method will work together with the `isChildRestricted()` method. This latter method indicates if children are allowed; while the former method puts restrictions, if necessary, on the child types. Keep in mind this is the base class and subclass will override when necessary.